compiler/src/dotty/tools/dotc/typer/RefChecks.scala

```diff
@@ -622,15 +625,12 @@ object RefChecks {
       if (member.isAnyOverride && !(clazz.thisType.baseClasses exists (hasMatchingSym(_, member)))) {
         // for (bc <- clazz.info.baseClasses.tail) Console.println("" + bc + " has " + bc.info.decl(member.name) + ":" + bc.info.decl(member.name).tpe);//DEBUG
 
-        val nonMatching = clazz.info.member(member.name).altsWith(alt => alt.owner != clazz && !alt.is(Final))
-        def issueError(suffix: String) =
-          ctx.error(i"$member overrides nothing$suffix", member.pos)
+        val nonMatching = clazz.info.member(member.name).altsWith(alt => alt.owner != clazz)
```

Comment:
The former error message listed only non-final methods with the same name, but I think it is useful to have even final methods in the "not eligible" list of methods so that one can easily see it as the modifiers are included.